### PR TITLE
[FLINK-11026][ES6] Rework creation of fat sql-client jars 

### PR DESCRIPTION
**Pull request by zentol**
Based on #7247.

## What is the purpose of the change

This PR is a PoC for reworking the packaging of jars specific to the sql-client (which basically are just fat-jars). Only the `flink-connector-elasticsearch6` module is covered here; if accepted the same principle should be applied to the kafka connectors (0.10, 0.11, 2) and all formats.

Instead of defining separate shade-plugin execution with a custom artifactSuffix this PR adds a dedicated `flink-sql-connector-elasticsearch6` module which only contains the packaging logic. This is a similar approach that we've already been using for `flink-shaded-hadoop2-uber`.

The main motivation for this is licensing; for accurate notice files it is necessary to be able to supply each artifact with distinct NOTICE files.

This cannot be done within a single module in a reasonable way. We would have to un-package each created jar, add the appropriate license files, and re-pack them again. We'd end up with tightly-coupled plugin definitions (since the names have to match!) and an overall more complicated (and slower!) build.

## Brief change log

* add new `flink-sql-connector-elasticsearch6` module containing the sql-client-specific shade-plugin configuration and apply the following modifications
  *  set `executionId` to `shade-flink`
  * disable `shadedArtifactAttached` so only a single jar is deployed
  * remove sql-jar suffix as it is no longer necessary
* remove sqlJars profile from `flink-connector-elasticsearch6`
* add `sqlJars` profile to `flink-connectors` to support skipping the creation of sql jars

## Verifying this change

Covered by sql-client E2E test.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)

## Documentation

I have not checked the documentation yet for references that would have to be changed.

### CODE REVIEW

**Comment by zentol** [comment]
There's an issue in this PR that causes `mvn install` to fail if the module was already built before. currently investigating what is causing this.

**Comment by zentol** [comment]
Issue should be fixed; it was the uncommon issue about shading of test-jars for modules that don't have any test classes.

File: flink-connectors/flink-connector-elasticsearch6/pom.xml [hunk 1]
```diff
@@ -174,95 +174,6 @@ under the License.
...
-											<shadedPattern>org.apache.flink.elasticsearch6.shaded.org.apache.logging</shadedPattern>
-										</relocation>
-										<relocation>
-											<pattern>com.fasterxml.jackson</pattern>
-											<shadedPattern>org.apache.flink.elasticsearch6.shaded.com.fasterxml.jackson</shadedPattern>
-										</relocation>
-									</relocations>
-									<!-- Relocate the table format factory service file. -->
-									<transformers>
-										<transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
```
Comment (tillrohrmann) [inline flink-connectors/flink-connector-elasticsearch6/pom.xml]: Why can we delete this transformer in the `flink-sql-connector-elasticsearch6` module?

File: flink-filesystems/flink-s3-fs-presto/pom.xml [hunk 2]
```diff
@@ -272,12 +272,6 @@ under the License.
 										<exclude>META-INF/LICENSE.txt</exclude>
 									</excludes>
 								</filter>
-								<filter>
-									<artifact>com.facebook.presto.hadoop:hadoop-apache2</artifact>
-									<includes>
-										<include>com/facebook/presto/hadoop/**</include>
-									</includes>
-								</filter>
```
Comment (tillrohrmann) [inline flink-filesystems/flink-s3-fs-presto/pom.xml]: Why are these changes necessary?

File: flink-connectors/flink-sql-connector-elasticsearch6/pom.xml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>shade-flink</id>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<shadeTestJar>false</shadeTestJar>
```
Comment (tillrohrmann) [inline flink-connectors/flink-sql-connector-elasticsearch6/pom.xml]: Why do we disable the `shadeTestJar` property? What is the default btw?

Comment (tillrohrmann) [reply flink-connectors/flink-sql-connector-elasticsearch6/pom.xml]: Forget this comment. You've already answered it here: https://github.com/apache/flink/pull/7251#issuecomment-445215947. Maybe the default is false and thus not needed here.

File: flink-connectors/flink-connector-elasticsearch6/pom.xml (hunk 1 above)
Comment (zentol) [reply flink-connectors/flink-connector-elasticsearch6/pom.xml]: pulled in through the `shade-flink` execution

File: flink-connectors/flink-sql-connector-elasticsearch6/pom.xml (hunk 3 above)
Comment (zentol) [reply flink-connectors/flink-sql-connector-elasticsearch6/pom.xml]: the `shade-flink` execution that we inherit from sets it to `true`.

File: flink-filesystems/flink-s3-fs-presto/pom.xml (hunk 2 above)
Comment (zentol) [reply flink-filesystems/flink-s3-fs-presto/pom.xml]: eh... whoops. these don't belong in here, reverting.

**Comment by zentol** [comment]
It might make sense to move the SQL classes, but I believe this is an orthogonal issue to how the modules are structured. This PR is only about making the required module-structure changes to support licensing; moving these classes however is more about how to structure our own code.

In the long run, particularly for consistency that is my main concern in this PR, I would very much like to have sql-client specific extensions (and possibly table-stuff in general) in separate modules for _all_ connectors/formats, regardless of packaging, but that's a separate issue.

**Comment by zentol** [comment]
@twalthr What's your take on this? I wouldn't want to change the SQL-packaging rules without anyone from the SQL side signing off on it.

**Comment by twalthr** [comment]
I just noticed that I have another Flink instance running on my machine. Will check the failure again.

**Comment by twalthr** [comment]
@zentol sorry for the false alarm. the changes look good to me (modulo my docs comment).

**Comment by zentol** [comment]
Don't scare me like that 🗡 . I've update the docs, rebased the PR and will merge it once travis is green.

**Pull request merged**